tests/stdlib/tparseutils.nim

```diff
@@ -57,6 +57,106 @@ block:
   doAssert parseOct("1464755", num8u) == 0
   doAssert num8u == 0
 
+  var num64i: int64
+  doAssert parseOct("1777777777777777777777", num64i) == 22
+  doAssert num64i == -1'i64
+
+  doAssert parseOct("2777777777777777777777", num64i) == 0
+
+  doAssert parseOct("0_77777777777777777777", num64i) == 22
```

Comment:
same comment as in a PR from a while ago, tests should be DRY-er because it makes it easier to read/write/increase coverage/improve the logic, eg:

```nim
  doAssert parseOct("0o1777777777777777777777", num64i) == 24
  doAssert num64i == -1'i64
```
=> something like this:
```nim
template parseOctCheck[T](a: string, expected: T) =
  var ret: T
  doAssert parseOct(a, ret) == a.len
  doAssert ret == expected

parseOctCheck "0o1777777777777777777777", -1'i64 # now it's a 1-liner
```

Reply:
fixed